src/arena.c

```diff
@@ -637,6 +637,14 @@ arena_decay_backlog_update_last(arena_decay_t *decay, extents_t *extents) {
 	size_t ndirty_delta = (ndirty > decay->nunpurged) ? ndirty -
 	    decay->nunpurged : 0;
 	decay->backlog[SMOOTHSTEP_NSTEPS-1] = ndirty_delta;
+
+	if (config_debug) {
+		if (ndirty > decay->peak_npages) {
+			decay->peak_npages = ndirty;
+		}
+		assert(decay->peak_npages >=
```

Comment:
This seems like an unnecessarily weak assertion.  Can we change ```peak_npages``` to be a ceiling, so that the assertion is relevant more of the time?

```C
if (config_debug) {
        if (ndirty > decay->ceil_npages) {
                decay->ceil_npages = ndirty;
        }
        size_t npages_limit =
            arena_decay_backlog_npages_limit(decay);
        assert(decay->ceil_npages >= npages_limit);
        if (decay->ceil_npages > npages_limit) {
                decay->ceil_npages = npages_limit;
        }
}
```